### ISSUE TITLE
Alt text for dog image accessibility

### ISSUE THREAD

**Issue by marcysutton** [jigsawpieces/dog-api-images#4]
This API is so awesome! One thing I wish it supported, though: programmatic alt text for images, so that people who can't see the screen get some kind of context when this API is used. It could also help if the images fail to load for some reason (like in my email client's default mode, where I have to hit "load all images"). 

The folks submitting images could write some killer captions for their pets, and the API could expose those as metadata. It wouldn't really help retroactively, so some alt text would have to be written for the existing collection of images. But it would be a great contribution to create a more accessible internet.

**Comment by jigsawpieces** [comment]
This is a great idea Marcy 🐶🐶🐶!

Atm the response object returned by the endpoint look like this for a single image file:

```
{
    "status": "success",
    "message": "image-1.jpg"
}
```

or this array for a list of images: 

```
{
    "status": "success",
    "message": [
        "image-1.jpg",
        "image-2.jpg",
        "image-3.jpg",
     ]
}
```

To make your proposed changes we could change the structure of response object so that it returns an object each time, e.g. 

```
{
    "status": "success",
    "message": {
        url: "image-1.jpg",
        altText: "A picture of a dog"
     }
}
```

and an array of objects for multiple images, e.g.

```
{
    "status": "success",
    "message": [
        {
             url: "image-1.jpg",
             altText: "A picture of a dog"
         }, 
        {
             url: "image-2.jpg",
             altText: "A picture of a bigger dog"
         }, 
        {
             url: "image-3.jpg",
             altText: "A picture of a small dog"
         }
     ]
}
```

Obviously this would be a significant change for the API and we would need to introduce some kind of versioning for the calls this in order to not create problems for existing users. @ElliottLandsborough what do you think?

**Comment by marcysutton** [comment]
All great ideas! I like your idea of defaulting to a breed name in the absence of alt text. Better than nothing, or a random filename.

**Comment by jigsawpieces** [comment]
Maybe @ElliottLandsborough as first iteration we set up the /alt suffix endpoint which defaults all alt text for every image as "An image of a {breed} dog" for now?

**Comment by marcysutton** [comment]
FYI: "image" is redundant information when used with an img tag, so I'd say just "{breed} dog"

**Comment by ElliottLandsborough** [comment]
New endpoints added - anything that returns images can have /alt appended to it.

https://dog.ceo/api/breeds/image/random/alt
https://dog.ceo/api/breeds/image/random/3/alt

https://dog.ceo/api/breed/hound/images/alt
https://dog.ceo/api/breed/hound/images/random/alt
https://dog.ceo/api/breed/hound/images/random/3/alt

https://dog.ceo/api/breed/hound/afghan/images/alt
https://dog.ceo/api/breed/hound/afghan/images/random/alt

P.S @jigsawpieces there is a typo here: https://dog.ceo/dog-api/documentation/sub-breed (afgan)

**Comment by marcysutton** [comment]
So rad this got added, thank you. 

I'm finally getting around to playing with it, and finding that altText is missing from https://dog.ceo/api/breeds/image/random/alt. It does work with https://dog.ceo/api/breeds/image/random/1/alt, so it's a fine workaround. But I thought I'd mention it here in case anyone else runs into the same issue.
